### PR TITLE
[Branch 2201.1.X] Add item group record support for invoices

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

This is to improve the invoice support and to add itemgroup records for the connector

Fixes #https://github.com/ballerina-platform/ballerina-extended-library/issues/473

One line release note: 
- Improved the netsuite invoice record CRUD operations with item-group record support
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X]  Unit tests 
- [X] Intergation tests

**Test Configuration**:
* Ballerina Version: Ballerina 2201.2.1 (Swan Lake Update 2)
* Operating System: Ubuntu
* Java SDK: JAVA 11

# Checklist:

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: .github/workflows/dev-stg-release.yml [hunk 1]
```diff
@@ -19,14 +19,14 @@ jobs:
         steps:
             - uses: actions/checkout@v2
             - name: Ballerina Build
-              uses: ballerina-platform/ballerina-action/@master
+              uses: ballerina-platform/ballerina-action/@2201.2.1
```
Comment (LakshanSS) [inline .github/workflows/dev-stg-release.yml:22]: For 2201.1.x branch, these workflows have to be updated to use `2201.1.1`

File: netsuite/Ballerina.toml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 [package]
-distribution = "2201.0.0"
+distribution = "2201.2.1"
 org = "ballerinax"
 name = "netsuite"
-version = "2.2.0"
+version = "3.0.1"
```
Comment (LakshanSS) [inline netsuite/Ballerina.toml:5]: For 2201.1.x branch, the version has to be bumped to `2.2.1`

File: netsuite/Dependencies.toml [hunk 3]
```diff
@@ -9,7 +9,7 @@ dependencies-toml-version = "2"
 [[package]]
 org = "ballerina"
 name = "auth"
-version = "2.2.0"
+version = "2.4.0"
```
Comment (LakshanSS) [inline netsuite/Dependencies.toml:12]: This Dependencies.toml shouldn't be updated for `2201.1.x` branch

**Comment by SanduDS** [comment]
closing as #https://github.com/ballerina-platform/module-ballerinax-netsuite/pull/81

**Comment by SanduDS** [comment]
closing as https://github.com/ballerina-platform/module-ballerinax-netsuite/pull/81

**Pull request closed without merging**